model/atmosphere/dycore/docs/_ext/icon4py_sphinx.py

```diff
@@ -0,0 +1,453 @@
+            call_string = self.get_next_method_call(source, source.find(docblock) + len(docblock))
+            next_method_info = self.get_method_info(call_string)
+            # and process the scidoc block
+            formatted_docblock = docblock.splitlines()
+            formatted_docblock = self.format_docblock(formatted_docblock, self.docblock_keyword)
+            formatted_docblock = self.make_header(next_method_info) + formatted_docblock
+            formatted_docblock = self.process_scidoc_lines(formatted_docblock, next_method_info)
+            formatted_docblock += self.SCIDOC_CODE_BLOCK_LINES + [" "*6 + line for line in call_string]
+
+            if formatted_docblock:
+                if i < len(docblocks)-1: # Add footer
+                    formatted_docblock += self.SCIDOC_FOOTER_LINES  
+                # add the processed docblock to the list
+                docblocks_list.append(formatted_docblock)
+
+        return docblocks_list
+    
+    def get_documentation_blocks(self, source: str, keyword: str) -> list[str]:
+        """
+        Extract blocks of comments from the source code that start with a specific keyword.
+
+        Args:
+            source: The source code as a string.
+            keyword: The keyword to look for in the comment blocks.
+
+        Returns:
+            A list of comment blocks that start with the specified keyword.
+        """
+        comment_blocks = []
+        in_block = False
+        for line in source.splitlines():
+            stripped_line = line.strip()
+            is_comment = stripped_line.startswith('#')
+            if stripped_line.startswith(f'# {keyword}:'):
+                comment_blocks.append(f'{line}')
+                in_block = True
+            elif in_block and is_comment:
+                comment_blocks[-1] += f'\n{line}'
+            in_block = is_comment and in_block # continue the block if the line is a comment (but don't start one if now with the keyword)
+        return comment_blocks
+
+    def format_docblock(self, docblock_lines: list[str], keyword: str) -> list[str]:
+        """
+        Format a block of documentation lines by cleaning up and removing keyword,
+        whitespace and empty lines.
+
+        Args:
+            docblock_lines: The lines of the documentation block to be formatted.
+            keyword: The keyword to look for and remove from the beginning of the docblock.
+
+        Returns:
+            The formatted documentation lines.
+        """
+        if docblock_lines[0].strip().startswith(f'# {keyword}:'):
+            docblock_lines.pop(0)  # remove the {keyword} prefix
+        while docblock_lines and docblock_lines[0].strip() == '#':
+            # Remove leading empty lines
+            docblock_lines.pop(0)
+        while docblock_lines and docblock_lines[-1].strip() == '#':
+            # Remove trailing empty lines
+            docblock_lines.pop(-1)
```

Comment:
white empty lines in the middle of the block are necessary (ReST) otherwise lists are not processed correctly.
`WARNING: Definition list ends without a blank line; unexpected unindent.`
I'll add the `assert` though